.eslintrc.cjs

```diff
@@ -35,6 +35,20 @@ module.exports = {
       'rules': {
         '@typescript-eslint/no-non-null-assertion': 2,
         '@typescript-eslint/no-explicit-any': 2,
+        '@typescript-eslint/member-delimiter-style': ['error', {
+          'multiline': {
+            'delimiter': 'semi',
+            'requireLast': true,
+          },
+          'singleline': {
+            'delimiter': 'semi',
+            'requireLast': true,
+          },
+        }],
+        'object-shorthand': 'error',
+        '@typescript-eslint/method-signature-style': ['error', 'method'],
```

Comment:
Why `method` over `property`? It seems like you get better strict checking with `property` as per https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/method-signature-style.md#enforces-using-a-particular-method-signature-syntax-method-signature-style

Reply:
"method" uniform the style with object-shorthand

Reply:
I think `object-shorthand` makes sense for JavaScript, but using `method-signature-style: property` fixes type issues that could lead to bugs.  So, even if `object-shorthand` looks better, I think the stronger typing is better overall.  Maybe the compromise is to say `object-shorthand: ['error', 'consistent']` so that it could be used for objects that do not have methods? Alternatively, we could leave it unset?

Reply:
> I think the stronger typing is better overall.

In my understanding, Both style contains same type information.

```typescript
// method-signature-style: method
export interface Data {
  CanAddle(s: string): boolean;
}

// method-signature-style: property
export interface Data {
  CanAddle: (s: string) => boolean;
}
```

Reply:
As per https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/method-signature-style.md#enforces-using-a-particular-method-signature-syntax-method-signature-style:
> A good practice is to use the TypeScript's strict option (which implies strictFunctionTypes) which enables correct typechecking for function properties only (method signatures get old behavior).
> 
> TypeScript FAQ:
> 
> A method and a function property of the same type behave differently. Methods are always bivariant in their argument, while function properties are contravariant in their argument under strictFunctionTypes.

I had to go read https://www.stephanboyer.com/post/132/what-are-covariance-and-contravariance to understand what covariance/contravariance/bivariance means.  I think the root of it is that it adds strictness when checking if two functions are type equivalent when the arguments to those functions are subtypes of each other.